Elasticsearch/build_elasticsearch.sh

```diff
@@ -9,290 +9,380 @@
-		fi
-		if [[ $(grep -rni "x-pack:plugin:ml" test_results.log) ]]; then
-			printf -- '****************************************************************************************************************************************\n'
-			printf -- '\n Few unit test case failures are observed on s390x as X-Pack plugin is not supported and Machine Learning is not available for s390x.\n'
-			printf -- '*****************************************************************************************************************************************\n'
-
-		fi
-	else
-    	 printf --  '\nCould not run test cases successfully\n'
-	fi
+function patch_for_7.1.1() {
+        # Applying patches
+        printf -- "Applying Patches for %s\n" "$PACKAGE_VERSION"
+        cd "${CURDIR}/elasticsearch"
+
+        curl -o patch_for_7_1_1.diff $REPO_URL/patch_for_7_1_1.diff
+        git apply patch_for_7_1_1.diff
+        rm -rf "${CURDIR}/patch_for_7_1_1.diff"
 
+        printf -- 'Patches applied!\n'
 }
 
-function startService() {
-	printf -- "\n\nstarting service\n"
-	cd "${CURDIR}/elasticsearch"
-	sudo tar -C /usr/share/ -xf distribution/archives/tar/build/distributions/elasticsearch-"${PACKAGE_VERSION}"-SNAPSHOT.tar.gz
-	sudo mv /usr/share/elasticsearch-"${PACKAGE_VERSION}"-SNAPSHOT /usr/share/elasticsearch
+function patch_for_7.0.1() {
+        # Applying patches
+        printf -- "Applying Patches for %s\n" "$PACKAGE_VERSION"
+        cd "${CURDIR}/elasticsearch"
 
-	if ([[ -z "$(cut -d: -f1 /etc/group | grep elastic)" ]]); then
-		printf -- '\nCreating group elastic\n'
-		sudo /usr/sbin/groupadd elastic # If group is not already created
+        curl -o patch_for_7_0_1.diff $REPO_URL/patch_for_7_0_1.diff
+        git apply patch_for_7_0_1.diff
+        rm -rf "${CURDIR}/patch_for_7_0_1.diff"
 
-	fi
-	sudo chown "$NON_ROOT_USER:elastic" -R /usr/share/elasticsearch
+        printf -- 'Patches applied!\n'
+}
 
-	#To access elastic search from anywhere
-	sudo ln -sf /usr/share/elasticsearch/bin/elasticsearch /usr/bin/
+function patch_for_7.0.0() {
+        # Applying patches
+        printf -- "Applying Patches for %s\n" "$PACKAGE_VERSION"
+        cd "${CURDIR}/elasticsearch"
 
-	# elasticsearch calls this file internally
-	sudo ln -sf /usr/share/elasticsearch/bin/elasticsearch-env /usr/bin/
+        curl -o patch_for_7_0_0.diff $REPO_URL/patch_for_7_0_0.diff
+        git apply patch_for_7_0_0.diff
+        rm -rf "${CURDIR}/patch_for_7_0_0.diff"
 
-	#Verify elasticsearch installation
-	if command -v "$PACKAGE_NAME" >/dev/null; then
-		printf -- "%s installation completed.\n" "$PACKAGE_NAME"
-	else
-		printf -- "Error while installing %s, exiting with 127 \n" "$PACKAGE_NAME"
-		exit 127
-	fi
+        printf -- 'Patches applied!\n'
+}
 
-	printf -- 'Service started\n'
+function build() {
+        #REPO_URL="https://raw.githubusercontent.com/linux-on-ibm-z/scripts/master/Elasticsearch/${PACKAGE_VERSION}"
```

Comment:
Removed the REPO_URL pointing to my REPO in the latest commit. And uncommented the official repo URL.